service/topology_state_machine.hh

```diff
@@ -28,6 +28,7 @@ namespace service {
 
 enum class node_state: uint16_t {
     none,                // the new node joined group0 but did not bootstraped yet (has no tokens and data to serve)
+    join_group0,         // the node was accepted but it needs to be added to group 0 before bootstrap/replace
```

Comment:
I suggested to add transition state, not a node state. I think it is more appropriate. We do not loose the information about what operation is ongoing.

Reply:
Sorry, I confused those two types of states.

@kbr-scylla WDYT? I remember asking you some time ago and you preferred a node state.

Reply:
I did, but I don't have any good arguments other than what you said: transition states are more for operations that change the entire cluster, while node states are for changing single node. OTOH you could say that the act of adding a new node has a significant effect, it affects feature calculation and so on, so maybe transition is indeed more appropriate.

 @gleb-cloudius why do you prefer transition state?

Reply:
Fine by me.

Reply:
Done, now `join_group0` is a transition state instead of node state.